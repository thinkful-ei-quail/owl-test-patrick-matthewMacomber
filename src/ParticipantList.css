.participantList {
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
}

.participant {
  display: flex;
}

.avatar {
  width: 10%;
  height: 10%;
  object-fit: cover;
  border-radius: 50%;
}

.info {
  display: flex;
  flex-direction: column;
}

.status {
  display: flex;
  justify-content: flex-start;
}

.status > div {
  margin-right: 10px;
}

.sessionOn {
  color: #64D481;
}

.sessionOff {
  color: lightgrey;
}

.statusText {
  color: #818994;
}

