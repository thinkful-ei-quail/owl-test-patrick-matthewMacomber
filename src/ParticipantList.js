import React from 'react';

function checkStatus(inSession, onStage) {
  if(!inSession) {
    return 'offline';
  }
  else if(onStage) {
    return 'on stage';
  }
  else {
    return 'in session';
  }
}

function ParticipantList(props) {
  return <div>
      {props.participants.map(par => {
        return  <section id={par.id} className="participant">
                  <img src={par.avatar} className="avatar" alt="Avatar" />
                  <div className="info">
                    <div className="name">
                      {par.name}
                    </div>
                    <div className="status">
                      <div className={par.inSession ? 'sessionOn' : 'sessionOff'}>
                        &#11044;
                      </div>
                      <div>
                        {checkStatus(par.inSession, par.onStage)}
                      </div>
                    </div>
                  </div>
                </section>
  })}
  </div>;
}

export default ParticipantList;