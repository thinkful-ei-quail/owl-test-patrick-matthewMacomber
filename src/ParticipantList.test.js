import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';

describe('Participant list testing', () => {
  const participants = [
    {
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    },
    {
        id: 2,
        name: 'Caty Flucker',
        avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
    },
    {
        id: 3,
        name: 'Axe Kubicka',
        avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
        inSession: false,
        onStage: false
    },
    {
        id: 4,
        name: 'Frank Runciman',
        avatar:
            'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
    },
    {
        id: 5,
        name: 'Ashla Attwool',
        avatar:
            'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }
];

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList participants={participants}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Smoke test
  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ParticipantList participants={participants}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});