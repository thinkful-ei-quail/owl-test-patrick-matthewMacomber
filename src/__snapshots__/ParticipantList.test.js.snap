// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant list testing renders the UI as expected 1`] = `<div />`;
